Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -732,14 +715,18 @@ class TrackStateProxy {
 
   /// Uncalibrated measurement in the form of a source link. Const version
   /// @return The uncalibrated measurement source link
-  const SourceLink& uncalibratedSourceLink() const;
+  [[deprecated("Use getUncalibratedSourceLink() instead")]] SourceLink
+  uncalibratedSourceLink() const;
```

Comment:
Good question, indeed. We've never really discussed a policy on that. 

Particularly in this case, if you keep the legacy version you increase the chance of someone ending up with a dangling reference. By forcing to change, the return type could be updated accordingly perhaps?

Reply:
I was debating whether to do this. Generally I think we'll have to transition to preserving backward compatibility more, and indeed this still compiles, but you're right it might be a problem. In theory, `const&` would prolong lifetime of a temporary returned here, but I've seen it crash, so yeah, might be better to remove the overload altogether.

Reply:
Ok, I dropped this function completely now.